.github/workflows/windows.yml

```diff
@@ -0,0 +1,96 @@
+name: Windows Tests
+on:
+  pull_request:
+    branches: [ '*' ]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+env:
+  GOPROXY: https://proxy.golang.org,direct
+  SDE_MIRROR_URL: "https://downloadmirror.intel.com/777395/sde-external-9.21.1-2023-04-24-win.tar.xz"
+  SDE_VERSION_TAG: sde-external-9.21.1-2023-04-24-win
+
+jobs:
+  MSVC-2019:
+    runs-on: temporary-performance-testing_windows-2019_8-core
+    steps:
+      - name: Git clone the repository
+        uses: actions/checkout@v3
+      - name: Build Windows Dependencies
+        run: |
+          choco install ninja --version 1.9.0.20190208 -y &&
+          choco install nasm --version 2.14.02 -y
+      - name: Run Windows Tests on MSVC-2019
+        run: |
+          .\tests\ci\run_windows_tests.bat "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" x64
+
+  MSVC-2022:
+    runs-on: temporary-performance-testing_windows-latest_8-core
+    steps:
+      - name: Git clone the repository
+        uses: actions/checkout@v3
+      - name: Build Windows Dependencies
+        run: |
+          choco install ninja --version 1.9.0.20190208 -y &&
+          choco install nasm --version 2.14.02 -y
+      - name: Run Windows Tests on MSVC-2022
+        run: |
+          .\tests\ci\run_windows_tests.bat "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64
+
+  SDE-64-bit:
+    # We have to run against windows-2019 due to an assembly intruction bug in MSVC2022's msvcrt.dll.
+    # The dll library tries to call vinsertf128 with memset, but this isn't available when testing
+    # against older processors with SDE.
```

Comment:
Do you have an internal issue tracking this and the an example of the failure you hit?

Reply:
Are you certain that it is a MSVC bug and not an issue with AWS-LC attempting to use the wrong assembly for a particular CPU?

Reply:
Saving the error here, how long to GHA logs stick around for?

```
crypto/mem_test (for CPU "icx")
crypto/mem_test (for CPU "knl")
TID 0 SDE-ERROR: Executed instruction not valid for specified chip (NEHALEM): 0x7ffc3a1d8080: vinsertf128 ymm0, ymm0, xmm0, 0x1
Image: C:\Windows\System32\msvcrt.dll+0x78080
Function: memset
Instruction bytes are: c4 e3 7d 18 c0 01 

crypto/mem_set_test (for CPU "nhm")
crypto/mem_set_test failed to complete: exit status 1
crypto/mem_test (for CPU "knm")
TID 0 SDE-ERROR: Executed instruction not valid for specified chip (WESTMERE): 0x7ffc3a1d8080: vinsertf128 ymm0, ymm0, xmm0, 0x1
Image: C:\Windows\System32\msvcrt.dll+0x78080
Function: memset
Instruction bytes are: c4 e3 7d 18 c0 01 

crypto/mem_set_test (for CPU "wsm")
crypto/mem_set_test failed to complete: exit status 1
crypto/mem_test (for CPU "tgl")
crypto/mem_set_test (for CPU "p4p")
crypto/mem_set_test (for CPU "mrm")
```

Reply:
* We synced up offline on this. I changed the wording to be a bit more "mild" while we investigate the issue. I'll be testing this edge case against a Windows instance in the meantime, but I think we can move on with adding coverage for SDE in Windows.
* GHA logs stick around for 90 days (which is the max limit).

Reply:
I ran [a test run with memset](https://github.com/aws/aws-lc/pull/1261/files) to verify that that's what failing specifically. It correlates with documentation that Windows 11 is phasing out some out some of the older processors, I've updated the wording with the link: 
* https://learn.microsoft.com/en-us/windows-hardware/design/minimum/supported/windows-11-supported-intel-processors